src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,143 @@
+            if (command.equals("delete")) {
+                Parser.checkInput(fullCommand, command, temp);
+                com = new DeleteCommand(Integer.parseInt(temp[1]));
+            } else if (command.equals("done")) {
+                checkInput(fullCommand, command, temp);
+                com = new DoneCommand(Integer.parseInt(temp[1]));
+            } else if (command.equals(("todo"))) {
+                checkInput(fullCommand, command, temp);
+                Task t = new Todo(fullCommand.substring(5));
+                com = new AddCommand(t);
+            } else if (command.equals("deadline")) {
+                checkInput(fullCommand, command, temp);
+                arr = fullCommand.split(" /by ");
+                String[] dt = arr[1].split(" ");
+                ld = LocalDate.parse(dt[0], inputDateFormat);
+                lt = LocalTime.parse(dt[1], inputTimeFormat);
+                Task d = new Deadline(arr[0].substring(9), ld, lt);
+                com = new AddCommand(d);
+            } else if (command.equals("event")) {
+                checkInput(fullCommand, command, temp);
+                arr = fullCommand.split(" /at ");
+                String[] dt = arr[1].split(" ");
+                ld = LocalDate.parse(dt[0], inputDateFormat);
+                lt = LocalTime.parse(dt[1], inputTimeFormat);
+                Task e = new Event(arr[0].substring(6), ld, lt);
+                com = new AddCommand(e);
+            } else if (command.equals("search")) {
+                checkInput(fullCommand, command, temp);
+                String description = fullCommand.substring(command.length() + 1);
+                com = new SearchCommand(LocalDate.parse(description, inputDateFormat));
+            } else if (command.equals("find")) {
+                checkInput(fullCommand, command, temp);
+                String description = fullCommand.substring(command.length() + 1);
+                com = new FindCommand(description);
+            } else if (command.equals("list")) {
+                com = new ListCommand();
+            } else if (command.equals("bye")) {
+                com = new ExitCommand();
+            } else {
+                com = new UnknownCommand();
+            }
+
+        } catch (DateTimeParseException e) {
+            throw new InvalidDateTimeException();
+        }
+
+        return com;
+    }
+
+    /**
+     * Takes in user command input and make sense of it to create the specific Task object and Command object.
+     *
+     * @param fullCommand This is the full command input of user.
+     * @param command This is the identifier for the full command.
+     * @param fullComArr This is the String array for the broken up full command.
+     * @return Nothing.
+     * @throws DukeException thrown when user input is missing information.
+     */
+    private static void checkInput(String fullCommand, String command, String[] fullComArr) throws DukeException {
+        if (command.equals("done")
```

Comment:
Since there are so many command strings, it may be best to store them as constants in an enum so they can be changed easily if needed

Example:
```
public enum CommandString {
    DONE("done"),
    DELETE("delete"),
    ..., // every command is inside
    SEARCH("search");

    private final String string;

    CommandString(String s) {
        string = s;
    }
}
```